synapse/push/push_tools.py

```diff
@@ -49,7 +50,33 @@ async def get_badge_count(store: DataStore, user_id: str, group_by_room: bool) -
 async def get_context_for_event(
     storage: StorageControllers, ev: EventBase, user_id: str
 ) -> Dict[str, str]:
-    ctx = {}
+    ctx: Dict[str, str] = {}
+
+    if ev.internal_metadata.outlier:
+        # We don't have state for outliers, so we can't compute the context
+        # except for invites with `invite_room_state` set..
```

Comment:
I was thinking about the possibility of falling back to the current state in the room as the latest room name is probably nicer than showing the `room_id`.

We can't do this because we don't want to leak the current state of the room? It feels like anyone getting a push notification would have access to the current state of the room :thinking:

Reply:
We only see outliers when the server isn't in the room, and so doesn't have the current state. If the server is in the room then the situation is different.

Reply:
Makes sense :+1:. I was thinking more outliers in the general sense like when events are authed from backfill, etc but those kind of things won't be sent to us from other servers or are relevant for pushers.